src/composables/latestTransactionList.ts

```diff
@@ -1,62 +1,214 @@
-import {
-  computed,
-  ref,
-} from 'vue';
-import { Encoded } from '@aeternity/aepp-sdk';
-import type { ITransaction } from '@/types';
-import { DASHBOARD_TRANSACTION_LIMIT } from '@/constants';
-import {
-  pipe,
+import { computed, ref, watch } from 'vue';
+import { isEqual, remove, uniq } from 'lodash-es';
+import type {
+  AccountAddress,
+  IAccount,
+  ICommonTransaction,
+  ITransaction,
+} from '@/types';
+import { PROTOCOLS, STORAGE_KEYS, TRANSACTION_CERTAINLY_MINED_TIME } from '@/constants';
+import { pipe, removeDuplicatedTransactions, sortTransactionsByDate } from '@/utils';
+import { ProtocolAdapterFactory } from '@/lib/ProtocolAdapterFactory';
+import { AE_MDW_TO_NODE_APPROX_DELAY_TIME } from '@/protocols/aeternity/config';
+
+import { useAccounts } from './accounts';
+import { useBalances } from './balances';
+import { useStorageRef } from './storageRef';
+import { useNetworks } from './networks';
+import { useAeSdk } from './aeSdk';
+import { useFungibleTokens } from './fungibleTokens';
+
+type AccountsTransactionList = Record<AccountAddress, ICommonTransaction[]>;
+
+let initialized = false;
+
+const removeDuplicatesAndSortTransactionsByDate = pipe([
   removeDuplicatedTransactions,
   sortTransactionsByDate,
-} from '@/utils';
-import { useTransactionTx } from './transactionTx';
-import { useTransactionList } from './transactionList';
+]);
+
+/**
+ * First page of the transactions done for each of the accounts
+ */
+const accountsTransactionsLatest = useStorageRef<AccountsTransactionList>(
+  {},
+  STORAGE_KEYS.transactionsLatest,
+);
+
+/**
+ * Pending transactions added manually right after transferring assets.
+ */
+const accountsTransactionsPending = useStorageRef<AccountsTransactionList>(
+  {},
+  STORAGE_KEYS.transactionsPending,
+);
+
+const areLatestTransactionsUpdating = ref(false);
+
+/**
+ * Latest and pending transactions combined for each account.
+ */
+const accountsTransactionsLatestAndPending = computed((): AccountsTransactionList => {
+  const accountAddresses: AccountAddress[] = uniq([
+    ...Object.keys(accountsTransactionsLatest.value),
+    ...Object.keys(accountsTransactionsPending.value),
+  ]);
+  return accountAddresses.reduce(
```

Comment:
Maybe we can use `Object.fromEntries` instead of reduce.

Reply:
This computed property was not used anymore so I removed it totally.